ligo/src/treasuryDAO.mligo

```diff
@@ -31,25 +21,22 @@ let treasury_DAO_proposal_check (params, extras : propose_params * contract_extr
   let min_xtz_amount = unpack_tez("min_xtz_amount", extras) in
   let max_xtz_amount = unpack_tez("max_xtz_amount", extras) in
 
-  let requred_token_lock = frozen_scale_value * proposal_size + frozen_extra_value in
-
-  if (params.frozen_token = requred_token_lock) && (proposal_size < max_proposal_size) then
-    let ts = unpack_transfer_type_list("transfers", params.proposal_metadata) in
-    let is_all_transfers_valid (is_valid, transfer_type: bool * transfer_type) =
-      match transfer_type with
-        | Token_transfer_type tt -> is_valid
-        | Xtz_transfer_type xt -> is_valid && min_xtz_amount <= xt.amount && xt.amount <= max_xtz_amount
-    in List.fold is_all_transfers_valid ts true
-  else
-    false
+  let required_token_lock = frozen_scale_value * proposal_size + frozen_extra_value in
+  let has_correct_token_lock =
+    (params.frozen_token = required_token_lock) && (proposal_size < max_proposal_size) in
 
+  let ts = unpack_transfer_type_list("transfers", params.proposal_metadata) in
+  let is_all_transfers_valid (is_valid, transfer_type: bool * transfer_type) =
+    match transfer_type with
+    | Token_transfer_type tt -> is_valid
+    | Xtz_transfer_type xt -> is_valid && min_xtz_amount <= xt.amount && xt.amount <= max_xtz_amount
+  in List.fold is_all_transfers_valid ts has_correct_token_lock
```

Comment:
Good point. I was assuming this fold would not have too many elements and so it would be ok to simplify the code a bit.

Reply:
Most likely that would indeed be the case, but there may be cases in which it can be quite long.

In any case if we can save some cost we should do it :sweat_smile: